openwisp_controller/config/tests/test_api.py

```diff
@@ -268,6 +268,19 @@ def test_template_create_no_org_api(self):
         self.assertEqual(r.status_code, 201)
         self.assertEqual(r.data['organization'], None)
 
+    def test_template_create_vpn_with_type_as_generic(self):
+        path = reverse('controller_config:api_template_list')
+        test_user = self._create_operator(organizations=[self._get_org()])
+        self.client.force_login(test_user)
+        vpn1 = self._create_vpn(name='vpn1', organization=self._get_org())
+        data = self._get_template_data.copy()
+        data['organization'] = self._get_org().pk
+        data['type'] = 'generic'
+        data['vpn'] = vpn1.id
+        response = self.client.post(path, data, content_type='application/json')
+        validation_msg = "To select a VPN, set the template type to 'VPN-client'"
+        self.assertIn(validation_msg, response.data['vpn'])
```

Comment:
it's always a good idea to assert for the status code as well, eg: `self.assertEqual(r.status_code, 400)`